core/src/main/java/com/linecorp/armeria/common/AttributeMap.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import java.util.Iterator;
+
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+
+/**
+ * Holds {@link Attribute}s which can be accessed via {@link AttributeKey}.
+ */
+@SuppressWarnings("ClassNameSameAsAncestorName")
+public interface AttributeMap extends io.netty.util.AttributeMap {
+
+    /**
+     * Returns the {@link Iterator} of all {@link Attribute}s this map contains.
+     */
+    Iterator<Attribute<?>> attrs();
```

Comment:
Should we add `AttributeMap ownAttrs()` so users can access attributes without the fallback to root? For some use cases I think this would be needed.

Reply:
Unfortunately, this proposed method should return `AttributeMap` I think, which then creates a typing conflict with `attrs()` which returns `Iterator`

Reply:
We could instead add `ownAttrs()`, `hasOwnAttrs()` and `ownAttr()` to `ClientRequestContext`?

Reply:
Yup that works well too

Reply:
Will add those three methods to `ClientRequestContext`, thanks!